### PR TITLE
implemented SchemaIO Transform for generating testing data

### PR DESCRIPTION
**Pull request by xianhualiu**

### CODE REVIEW

File: syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/datagenerator/DataGeneratorSchemaTransform.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import org.apache.beam.sdk.transforms.PeriodicImpulse;
+import org.apache.beam.sdk.transforms.Reshuffle;
+import org.apache.beam.sdk.transforms.SimpleFunction;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionRowTuple;
+import org.apache.beam.sdk.values.Row;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+
+public class DataGeneratorSchemaTransform implements SchemaTransform, Serializable {
```
Comment (bvolpato) [inline syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/datagenerator/DataGeneratorSchemaTransform.java:37]: Recommend adding some javadoc here

File: syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/datagenerator/DataGeneratorSchemaTransform.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    @Override
+    public List<Row> apply(Instant input) {
+      final Schema schema = Schema.parse(this.avroSchemaString);
+      long ordinal = (input.minus(Duration.millis(startTime.getMillis())).getMillis() / 1000);
+      return LongStream.range(ordinal * recordsPerSecond, (ordinal + 1) * recordsPerSecond)
+          .mapToObj(intVal -> RecordCreator.createRowRecord(schema, beamSchema))
+          .collect(Collectors.toList());
+    }
```
Comment (bvolpato) [inline syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/datagenerator/DataGeneratorSchemaTransform.java:104]: This looks complicated. Can we add a commend explain what it's doing?

File: syndeo-template/src/test/java/com/google/cloud/syndeo/transforms/datagenerator/DataGeneratorSchemaTransformTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    DataGeneratorSchemaTransformProvider provider =
+        (DataGeneratorSchemaTransformProvider)
+            ProviderUtil.getProvider("syndeo:schematransform:com.google.cloud:data_generator:v1");
+    DataGeneratorSchemaTransform schemaTransform =
+        (DataGeneratorSchemaTransform)
+            provider.from(
+                DataGeneratorSchemaTransformConfiguration.builder()
+                    .setSchema(schema)
+                    .setSecondsToRun(5l)
+                    .setRecordsPerSecond(100l)
```
Comment (bvolpato) [inline syndeo-template/src/test/java/com/google/cloud/syndeo/transforms/datagenerator/DataGeneratorSchemaTransformTest.java:46]:
```suggestion
                    .setSecondsToRun(5L)
                    .setRecordsPerSecond(100L)
```

**Review by bvolpato — APPROVED** [review]
No concerns with the template, just left a few nit comments

File: syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/datagenerator/DataGeneratorSchemaTransform.java (hunk 2 above)
Comment (xianhualiu) [reply syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/datagenerator/DataGeneratorSchemaTransform.java:158]: added comments and exmaples

**Pull request merged**